draft-thomson-webpush-vapid.md

```diff
@@ -132,211 +134,272 @@ The words "MUST", "MUST NOT", "SHOULD", and "MAY" are used in this document.
 
-A push service that supports application server self-identification requests a
-client certificate from application servers.  The client certificate is
-requested during the TLS [RFC5246] handshake.
+ * An "exp" (Expiry) claim MUST be included with the time after which the token
+   expires.  This limits the time that a token over which a token is valid.  An
+   "exp" claim MUST NOT be more than 24 hours from the time of the request.
 
-An application server that does not have a client certificate offers no
-certificate in response; an application server that wishes to self-identify
-includes a certificate.
+This JWT is included in an Authorization header field, using an auth-scheme of
+"WebPush".  A push service MAY reject a request with a 401 (Unauthorized) status
+code [RFC7235] if the JWT signature or its claims are invalid.
 
-The certificate that the application server offers SHOULD be self-signed (see
-Section 3.2 of [RFC5280]).  The certificate MAY contain an alternative name
-extension (Section 4.2.1.6 of [RFC5280]) that includes contact information.  Of
-the available options, an email address using the `rfc822Name` form or an HTTP
-[RFC7230] (or HTTPS [RFC2818]) `uniformResourceIdentifier` SHOULD be included,
-though including other options are not prohibited.
+The JWT MUST use a JSON Web Signature (JWS) [RFC7515].  The signature MUST use
+ECDSA on the NIST P-256 curve [FIPS186], that is "ES256" [RFC7518].
 
-The offered end-entity certificate or the public key it contains becomes an
-identifier for the application server.  Push services are able to reduce the
-data they retain for an application server, either by extracting important
-information like the subject public key information (SPKI), by hashing, or a
-combination.  Of course, a push service might choose to ignore the provided
-information.
 
-To avoid requesting certificates from user agents, it might be necessary to
-serve requests from user agents and requests from application servers on
-different hostnames or port numbers.
+## Application Server Contact Information
 
+If the application server wishes to provide the JWT can include an "iss"
+(Issuer) claim.  The "iss" claim MAY include a contact URI for the application
+server; either a "mailto:" (email) [RFC6068] or an "https:" [RFC2818] SHOULD be
+provided.
 
-## Server-Vended Tokens {#server-token}
 
-In this model, the push service vends a token to each application server that
-requests it.  That token is kept secret and used as the basis for
-authentication.
+## Example
 
-This doesn't address the need for contact information, but it addresses the need
-for a consistent application server identifier.
+An application server requests the delivery of a push message as described in
+[I-D.ietf-webpush-protocol].  If the application server wishes to self-identify,
+it includes an Authorization header field with credentials that use the
+"WebPush" authentication scheme {{auth}} and a Crypto-Key header field that
```

Comment:
It wasn't clear that RFC 6750 wasn't exclusively for OAuth2.  I meant to ping Mike Jones about it, but haven't.  I will do that now.